usl_pipeline/usl_lib/usl_lib/chunkers/elevation_chunkers.py

```diff
@@ -0,0 +1,46 @@
+from osgeo import gdal
+import typing
+
+from usl_lib.readers import elevation_readers
+from usl_lib.shared import suppliers
+
+
+def split_geotiff_into_chunks(
+    elevation_file_path: str,
```

Comment:
expecting a `typing.BinaryIO` rather than a file path, as we do in `read_from_geotiff`, can be more flexible. For instance, we could write the tests using BytesIO objects rather than tempfiles.

Reply:
gdal.Open() can only work with string path: https://gdal.org/api/python/raster_api.html#osgeo.gdal.Open

Reply:
If I try to pass file reader I see the following error:
```
usl_pipeline/usl_lib/usl_lib/chunkers/elevation_chunkers.py:32: in split_geotiff_into_chunks
    ds = gdal.Open(file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<_io.BufferedReader name='/tmp/tmp6p9b5_7d'>,)

    def Open(*args) -> "GDALDatasetShadow *":
        r"""Open(char const * utf8_path, GDALAccess eAccess=GA_ReadOnly) -> Dataset"""
>       return _gdal.Open(*args)
E       RuntimeError: not a string
```

Reply:
cool -- let's stick with file path strings, then.